compiler/next/include/chpl/uast/FunctionDecl.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_FUNCTIONDECL_H
+#define CHPL_UAST_FUNCTIONDECL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Function.h"
+
+#include <cassert>
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a function declaration
+  E.g. here are some examples
+
+  \rst
+  .. code-block:: chapel
+
+      proc f(arg) { }
+
+      proc g(x: int = 32) where something() { }
+  \endrst
+
+  Each of these is a FunctionDecl that refers to a Function Sym.
+ */
+class FunctionDecl final : public Decl {
+ private:
+  FunctionDecl(owned<Function> function)
+    : Decl(asttags::FunctionDecl, std::move(function)) {
+  }
+  bool contentsMatchInner(const ASTNode* other) const override;
+  void markUniqueStringsInner(Context* context) const override;
+
+ public:
+  ~FunctionDecl() override = default;
+
+  // if isMethod is true, the first formal is the 'this' formal
```

Comment:
I don't see isMethod in the argument list for this builder (I'm assuming because methods aren't supported yet).  Just pointing that out

Reply:
Actually it's because `owned<FormalDecl> receiver,` indicates if it is a method. The comment is out of date & I'll remove it.

I don't have much testing for methods yet, so they're not fully formed yet, it's true.